gidgethub/abc.py

```diff
@@ -37,11 +40,25 @@ def __init__(
         requester: str,
         *,
         oauth_token: Opt[str] = None,
+        app_id: Opt[str] = None,
+        private_key: Opt[str] = None,
         cache: Opt[CACHE_TYPE] = None,
         base_url: str = sansio.DOMAIN,
     ) -> None:
+        if all(_ is not None for _ in (oauth_token, app_id, private_key)):
+            raise ValueError(
+                "Cannot pass oauth_token at the same time as app_id and "
```

Comment:
The error message is a little too detailed. I would suggest simplifying this. The details about which one to choose should be clarified in the documentation (abc.rst)

Perhaps:

"Cannot pass oauth_token if app_id and private_key are also passed."

and explain it in the docs.